File: src/main/java/no/novari/flyt/history/validation/MinTimestampBeforeMaxTimestampValidator.java
package no.novari.flyt.history.validation;

import no.novari.flyt.history.model.time.ManualTimeFilter;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import java.util.Optional;

public class MinTimestampBeforeMaxTimestampValidator implements ConstraintValidator<MinTimestampBeforeMaxTimestamp, ManualTimeFilter> {

    @Override
    public boolean isValid(ManualTimeFilter manualTimeFilter, ConstraintValidatorContext constraintValidatorContext) {
        return Optional.ofNullable(manualTimeFilter.getMin())
                .flatMap(min -> Optional.ofNullable(manualTimeFilter.getMax()).map(min::isBefore))
                .orElse(true);
    }

}

File: src/main/kotlin/no/novari/flyt/history/validation/MinTimestampBeforeMaxTimestampValidator.kt
package no.novari.flyt.history.validation

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext
import no.novari.flyt.history.model.time.ManualTimeFilter

class MinTimestampBeforeMaxTimestampValidator : ConstraintValidator<MinTimestampBeforeMaxTimestamp, ManualTimeFilter> {
    override fun isValid(
        manualTimeFilter: ManualTimeFilter?,
        context: ConstraintValidatorContext?,
    ): Boolean {
        val min = manualTimeFilter?.min ?: return true
        val max = manualTimeFilter.max ?: return true
        return min.isBefore(max)
    }
}
